From a8da1e54128213658fd9221128828c6c7e15d24e Mon Sep 17 00:00:00 2001
From: stephan-xyz <stephan@zehrer.xyz>
Date: Tue, 27 Oct 2026 11:47:42 +0100
Subject: [PATCH] Revert relibc patch 0004 (x1/x2 swap in the signal entry)

Negative control for tools/sigregs: brings back the swap of x1 and x2 when
a signal handler returns.
---
 redox-rt/src/arch/aarch64.rs | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/redox-rt/src/arch/aarch64.rs b/redox-rt/src/arch/aarch64.rs
index a489b05..7e8d0f7 100644
--- a/redox-rt/src/arch/aarch64.rs
+++ b/redox-rt/src/arch/aarch64.rs
@@ -369,10 +369,8 @@ asmfunction!(__relibc_internal_sigentry: ["
     mrs x3, nzcv
     stp x2, x3, [sp, #-16]!
 
-    // x2 goes below x1 (ArchIntRegs order, and `ldp x2, x1` below), like
-    // the other pairs.
     ldp x2, x3, [x0, #{tcb_sa_off} + {sa_tmp_x1_x2}]
-    stp x3, x2, [sp, #-16]!
+    stp x2, x3, [sp, #-16]!
     ldp x3, x4, [x0, #{tcb_sa_off} + {sa_tmp_x3_x4}]
     stp x4, x3, [sp, #-16]!
     ldp x5, x6, [x0, #{tcb_sa_off} + {sa_tmp_x5_x6}]
-- 
2.54.0 (Apple Git-157)

